src/domains/core/screens/RootStackNavigator.tsx

```diff
@@ -0,0 +1,23 @@
+import { createStackNavigator, StackNavigationProp } from '@react-navigation/stack';
+import React, { FC } from 'react';
+import { BottomTabNavigator } from 'src/domains/core/screens/BottomTabNavigator';
+import { ModalTutorialScreen } from 'src/domains/modalTutorial/ModalTutorialScreen';
+
+export type RootStackNavigatorParams = {
+  BottomTabNavigator: undefined;
+  ModalTutorialScreen: undefined;
+};
+
+const Stack = createStackNavigator<RootStackNavigatorParams>();
+
+export type RootStackNavigationProp<ScreenName extends keyof RootStackNavigatorParams> = StackNavigationProp<
```

Comment:
We can compose a few different types of these `NavigationProp` types to make it simpler to pull in.